File: LogicalThinkingPatterns/AlphaTrianglePattern/AlphaTriangle.py
class AlphaTriangle:
    def printPattern(self, n):
        for i in range(0, n):
            start = ord('A') + n - i - 1
            for ch in range(start, ord('A') + n):
                print(chr(ch)+" ", end="")
            print()

    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

alphaTriangle = AlphaTriangle()
alphaTriangle.run()

File: LogicalThinkingPatterns/AlphaTrianglePattern/AlphaTriangle.ts
class AlphaTriangle {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let start: number = 65 + n - i - 1;
      let pattern: string = "";
      for (let j = start; j <= 65 + n - 1; j++) {
        pattern += String.fromCharCode(j) + " ";
      }
      console.log(pattern);
    }
  }
}

const alphaTriangle = new AlphaTriangle();
alphaTriangle.printPattern(5);
